exercises/raindrops/example.scm

```diff
@@ -1,26 +1,15 @@
-(define-module (raindrops)
-  #:export (convert))
+(import (rnrs (6)))
 
-(define prime-factors
-  (lambda (n)
-    (let loop ((n n) (m 2) (factors '()))
-      (cond ((= n 1) factors)
-            ((= 0 (modulo n m)) (loop (/ n m) 2 (cons m factors)))
-            (else (loop n (+ 1 m) factors))))))
+(load "test.scm")
+
+(define (convert number)
+  (let* ((rules '((3 . "Pling") (5 . "Plang") (7 . "Plong")))
+	 (helper (lambda (k.w word)
+		   (if (zero? (modulo number (car k.w)))
+		       (string-append (cdr k.w) word)
+		       word)))
+	 (result (fold-right helper "" rules)))
+    (if (equal? "" result)
+	(number->string number)
+	result)))
 
-(define convert
-  (lambda (n)
-    (let ((drops "")
-          (factors (filter (lambda (v)
-                             (and (>= v 3) (<= v 7)))
-                           (prime-factors n))))
-      (if (not (null? factors))
-          (begin
-            (when (member 3 factors)
-              (set! drops (string-append drops "Pling")))
-            (when (member 5 factors)
-              (set! drops (string-append drops "Plang")))
-            (when (member 7 factors)
-              (set! drops (string-append drops "Plong"))))
-          (set! drops (string-append drops (number->string n))))
```

Comment:
wow

Reply:
This is the happiest section of a diff I have ever seen 🤣
Wow indeed!